src/device-manager/java/src/nl/Weave/DataManagement/GenericTraitUpdatableDataSink.java

```diff
@@ -0,0 +1,78 @@
+/*
+
+    Copyright (c) 2019 Google, LLC.
+    All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Represents GenericTraitUpdatableDataSink Interface.
+ */
+
+package nl.Weave.DataManagement;
+
+import android.os.Build;
+import android.util.Log;
+
+import java.math.BigInteger;
+import java.util.EnumSet;
+import java.util.Random;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
+
+public interface GenericTraitUpdatableDataSink
+{
+    public void setSigned(String path, int value, boolean isConditional);
+    public void setSigned(String path, long value, boolean isConditional);
+    public void setSigned(String path, BigInteger value, boolean isConditional);
+    public void setUnsigned(String path, int value, boolean isConditional);
+    public void setUnsigned(String path, long value, boolean isConditional);
+    public void setUnsigned(String path, BigInteger value, boolean isConditional);
+    public void set(String path, double value, boolean isConditional);
+    public void set(String path, boolean value, boolean isConditional);
+    public void set(String path, String value, boolean isConditional);
+    public void set(String path, byte[] value, boolean isConditional);
+    public void setNull(String path, boolean isConditional);
```

Comment:
Was just thinking about this a bit more. Should we remove the setNull() function entirely, and use wrapper classes on the set functions? e.g.

setSigned(String path, Integer value, boolean isConditional);
setSigned(String path, Long value, boolean isConditional);
set(String path, Double value, boolean isConditional);
set(String path, String value, boolean isConditional);

That way, you can use the same function, just providing null as the argument. It does incur some auto-boxing overhead but I don't think that's really an issue here.

Reply:
I think that's fine idea, but I would prefer to keep setNull() as well.

Reply:
I am keeping setNull here, thanks